bentoml/_internal/io_descriptors/file.py

```diff
@@ -155,6 +165,39 @@ async def to_http_response(
             res = Response(body)
         return res
 
+    async def to_proto(self, obj: FileType) -> pb.File:
+        from bentoml.grpc.utils.mapping import mimetype_to_filetype_pb_map
+
+        if self._mime_type.startswith("multipart"):
+            raise UnprocessableEntity(
+                "'multipart' Content-Type is not yet supported for parsing files in gRPC. Use Multipart() instead."
+            )
+
+        if isinstance(obj, bytes):
+            body = obj
+        else:
+            body = obj.read()
+
+        try:
+            kind = mimetype_to_filetype_pb_map()[self._mime_type]
```

Comment:
For now, but we can add more later. It is pretty easy to add as well.

Reply:
That's not my issue; why are we limiting the kinds of files users can upload? We can't possibly have a list of every possible file type users may want to upload to APIs, so I think this doesn't make sense. There should at least be some way to allow for a custom mime type.

Reply:
Another solution is to make this mime type instead of a enum we can just make as a text, and then user can put in whatever mimetype they want.

Reply:
I think we should have a `pb.File.FILE_TYPE_CUSTOM` and then provide an additional field for it. I think having enum is pretty nice since they can choose and use those enum from the generated stubs directly.

Reply:
https://github.com/bentoml/BentoML/issues/2992 tracked here.